src/components/Layout/ChipBase.js

```diff
@@ -1,48 +1,94 @@
-            count: items.length,
+const getChipSuffix = ({
+    dimension,
+    conditions = {},
+    items = [],
+    metadata,
+}) => {
+    const { id, dimensionType, optionSet, valueType } = dimension
+
+    if (
+        (id === DIMENSION_ID_ORGUNIT ||
+            dimensionType === DIMENSION_TYPE_PERIOD) &&
+        !items.length
+    ) {
+        return ''
+    }
+
+    const all = i18n.t('all')
+
+    if (!conditions.condition && !conditions.legendSet && !items.length) {
+        return `: ${all}`
+    }
+
+    const conds = getConditions({ conditions, metadata, dimension })
+
+    if (
+        (valueType === VALUE_TYPE_TRUE_ONLY &&
+            conds.length === VALUE_TYPE_TRUE_ONLY_NUM_OPTIONS) ||
+        (valueType === VALUE_TYPE_BOOLEAN &&
+            conds.length === VALUE_TYPE_BOOLEAN_NUM_OPTIONS)
+    ) {
+        return `: ${all}`
+    }
+
+    const numberOfConditions = conds.length
+
+    let suffix = ''
+    if (optionSet || items.length) {
+        const selected = items.length || numberOfConditions
+
+        suffix = i18n.t('{{count}} selected', {
+            count: selected,
+            defaultValue: '{{count}} selected',
+            defaultValue_plural: '{{count}} selected',
         })
     } else if (numberOfConditions) {
-        itemsLabel = i18n.t('{{count}} conditions', {
+        suffix = i18n.t('{{count}} conditions', {
             count: numberOfConditions,
             defaultValue: '{{count}} condition',
             defaultValue_plural: '{{count}} conditions',
         })
     }
-    return itemsLabel ? `: ${itemsLabel}` : ''
+    return suffix ? `: ${suffix}` : ''
 }
 
-export const ChipBase = ({
-    dimensionName,
-    dimensionType,
-    items,
-    numberOfConditions,
-}) => (
-    <div className={cx(styles.chipBase)}>
-        <div className={styles.leftIcon}>
-            <DimensionIcon dimensionType={dimensionType} />
+export const ChipBase = (props) => {
+    const { name, dimensionType, stageName } = props.dimension
+    return (
+        <div className={cx(styles.chipBase)}>
+            <div className={styles.leftIcon}>
+                <DimensionIcon dimensionType={dimensionType} />
+            </div>
+            <span className={styles.label}>
+                <span className={styles.primary}>{name}</span>
+                {stageName && (
+                    <span className={styles.secondary}>{stageName}</span>
+                )}
+            </span>
+            <span className={styles.suffix}>{getChipSuffix(props)}</span>
         </div>
-        <span className={styles.label}>{dimensionName}</span>
-        <span className={styles.label}>
-            {renderChipLabelSuffix(items, numberOfConditions)}
-        </span>
-    </div>
-)
+    )
+}
 
 ChipBase.propTypes = {
-    dimensionName: PropTypes.string,
-    dimensionType: PropTypes.string,
-    items: PropTypes.array,
-    numberOfConditions: PropTypes.number,
+    dimension: PropTypes.object,
```

Comment:
missing prop types for `items, conditions, metadata`

Reply:
oops how did that sneak by